source/credentials_provider_profile.c

```diff
@@ -359,40 +372,44 @@ struct aws_credentials_provider *aws_credentials_provider_new_profile(
     struct aws_string *config_file_path = NULL;
     struct aws_string *profile_name = NULL;
 
-    credentials_file_path = aws_get_credentials_file_path(allocator, &options->credentials_file_name_override);
-    if (!credentials_file_path) {
-        AWS_LOGF_ERROR(
-            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
-            "static: Profile credentials parser failed resolve credentials file path");
-        goto on_finished;
-    }
-
-    config_file_path = aws_get_config_file_path(allocator, &options->config_file_name_override);
-    if (!config_file_path) {
-        AWS_LOGF_ERROR(
-            AWS_LS_AUTH_CREDENTIALS_PROVIDER, "static: Profile credentials parser failed resolve config file path");
-        goto on_finished;
-    }
-
     profile_name = aws_get_profile_name(allocator, &options->profile_name_override);
     if (!profile_name) {
         AWS_LOGF_ERROR(
             AWS_LS_AUTH_CREDENTIALS_PROVIDER, "static: Profile credentials parser failed to resolve profile name");
         goto on_finished;
     }
-    config_profiles = aws_profile_collection_new_from_file(allocator, config_file_path, AWS_PST_CONFIG);
-    credentials_profiles = aws_profile_collection_new_from_file(allocator, credentials_file_path, AWS_PST_CREDENTIALS);
 
-    if (!(config_profiles || credentials_profiles)) {
-        AWS_LOGF_ERROR(
-            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
-            "static: Profile credentials parser could not load or parse"
-            " a credentials or config file.");
-        goto on_finished;
-    }
+    merged_profiles = aws_profile_collection_acquire(options->profile_collection_cached);
+    if (!merged_profiles) {
```

Comment:
trivial: suggestion is more lines of code, but IMO reads more clearly to a newcomer
```suggestion
    if (options->profile_collection_cached) {
        /* Use cached profile collection */
        merged_profiles = aws_profile_collection_acquire(options->profile_collection_cached);
    } else {
        /* Load profile collection from files */

```